### ISSUE TITLE
Alphabetical filter and search view for person

### ISSUE THREAD

**Issue by danahertzberg** [az-digital/az_quickstart#1869]
**Do not disclose security vulnerabilities here, please email us instead: 
az-digital-security@list.arizona.edu**

## Motivation
I want to display people in a row view with alphabetical groupings and search by name filter.

### Is your feature request related to a problem? Please describe.
Alpha index view does not allow for person row display.

## Proposed Resolution
Add 2 new view displays to person view.

### Describe alternatives you've considered
Updating alpha index view won't work because that template is hard-coded.

### Roles and Permissions considerations
A clear and concise description of how each of the following roles would be impacted by this change:
- Content editor
- Content administrator
- Administrator

## Additional context
- Content editor workaround: https://live-azs-optics.pantheonsite.io/osc-arizona/directory
- Custom view: https://live-azs-psychology.pantheonsite.io/faculty

**Comment by natebyerley** [comment]
@danahertzberg - curious what horizontal finder style updates are?  This is one other view that would prevent people from doing odd workarounds.  I think creation of this view might also allow us to decustomize azs-psychology, and help azs-optics with their manual alphabetical view maintenance: https://www.optics.arizona.edu/our-college/full-directory

**Comment by danahertzberg** [comment]
@natebyerley Great question. This alphabetical filter is notoriously challenging to work with unfortunately. In the AZ Finder, we have the styling done for the vertical filters (usually) on the left of the page. But we also have an opportunity to add styling for filters that are at the top of a page (horizontal filters). This styling is not yet complete, but if we pair it with the alphabetical sort field found with this view, then it might make the functionality more flexible. Thank you for adding the examples so we know what to design for.

**Comment by BaoNguyen09** [comment]
Do you want this new view to only be enabled for specific page like "directory" or it can be added as a new display of az_person view?

<img src="https://github.com/user-attachments/assets/2816b557-e37a-476a-a5b9-e4c7ee0a4223" width="30%" />

Another option is to include these 2 views, as part of the existing "Row Page" view.

**Comment by natebyerley** [comment]
@BaoNguyen09 - I would not restrict it to /directory. I would definitely make it a standalone View of az_person. 

Not quite clear what you mean by your last suggestion: "Another option is to include these 2 views, as part of the existing "Row Page" view."

**Comment by danahertzberg** [comment]
I agree with Nate on not restricting it/overriding existing views. And I'm also confused by the last comment. Some more clarity would be helpful. Thanks!

**Comment by BaoNguyen09** [comment]
> Another option is to include these 2 views, as part of the existing "Row Page" view.
For this I mean, modifying the existing Row Page view to include the alphabetical groupings and search by name filter. But since, we all agree to add this one as a seperate view, we can go with that

**Comment by tadean** [comment]
In the workshop we identified a few notes:

- In a proposed glossary mode, it would be nice if the glossary list remained a non-sparse glossary rather than a sparse glossary. By default, glossaries are sparse and only show the letters that actually exist.
- It would potentially be nice if the index letter field could be maintained because this allows for a given entity to appear under multiple headings
- It seems that the glossary mode indexing removes accent letters when actually doing the search somehow, which seems to be what we'd want
- Rendered output being used for grouping can potentially cause people to appear in a separate group with an accented character
- Rendered glossary can sometimes cause the glossary letter to be the accented one, if they're the first one that appears in the database. Strangely, this doesn't seem to impact searching

**Comment by danahertzberg** [comment]
I agree with the non-sparse glossary goal. Could also consider a dropdown (for mobile or all screen sizes) for the letter selection.

**Comment by tadean** [comment]
@BaoNguyen09 and I met on friday to talk about initial goals on a exploratory draft about this. We identified that some first steps in determining if this is feasible might be:

- Create a glossary view that builds its glossary from the alphabetical index letter field rather than the title
- Attempt to use ` hook_views_post_execute` (or a potentially similar hook) to manually add letter rows to the glossary attachment which are missing from the glossary because they have no results

**Comment by tadean** [comment]
> I agree with the non-sparse glossary goal. Could also consider a dropdown (for mobile or all screen sizes) for the letter selection.

A dropdown seems like it would be possible.

**Comment by tadean** [comment]
I've opened a related task #4338 at @BaoNguyen09 's recommendation.
